src/data/List.js

```diff
@@ -1176,7 +1176,7 @@ function getRawIndexWithoutIndices(idx) {
 }
 
 function getRawIndexWithIndices(idx) {
-    if (idx < this._count && idx >= 0) {
```

Comment:
I think we should use `idx < this._count` here, otherwise it will get incorrect result.
The #10595 is probably a bug of axisPointer itself.
I am debugging it.

Reply:
I've known the reason: 
when the axis is updated by `dataZoom` or any other approaches like setOption, the `axisPonterModel` does not get the chance to update itself. 
But to fix this problem, the implementation of `axisPointer` has to be modified a lot. (And by some historical reason, the current implementation of `axisPointer` is bug-prone).
I will think this through and make some change of it gingerly ...